bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/discovery/RadioThermostatDiscoveryService.java

```diff
@@ -0,0 +1,301 @@
+            logger.debug("Joined UPnP Multicast group on Interface: {}", ni.getName());
+            byte[] requestMessage = RADIOTHERMOSTAT_DISCOVERY_MESSAGE.getBytes("UTF-8");
+            DatagramPacket datagramPacket = new DatagramPacket(requestMessage, requestMessage.length, m, port);
+            socket.send(datagramPacket);
+            try {
+                // Try to ensure that joinGroup has taken effect. Without this delay, the query
+                // packet ends up going out before the group join.
+                Thread.sleep(1000);
+
+                socket.send(datagramPacket);
+
+                byte[] buf = new byte[4096];
+                DatagramPacket packet = new DatagramPacket(buf, buf.length);
+
+                try {
+                    while (!Thread.interrupted()) {
+                        socket.receive(packet);
+                        String response = new String(packet.getData());
+                        logger.debug("Response: {} ", response);
+                        if (response.contains(SSDP_MATCH)) {
+                            logger.debug("Match: {} ", response);
+                            parseResponse(response);
+                        }
+                    }
+                    logger.debug("Bridge device scan interrupted");
+                } catch (SocketTimeoutException e) {
+                    logger.debug(
+                            "Timed out waiting for multicast response. Presumably all devices have already responded.");
+                }
+            } finally {
+                socket.leaveGroup(m);
+                socket.close();
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.debug("got exception: {}", e.getMessage());
+        }
+        return;
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and scans them for the
+     * search keywords. The search is not case sensitive.
+     *
+     * @param socket
+     *            The socket where the answers arrive.
+     * @param keywords
+     *            The keywords to be searched for.
+     * @return
+     * @throws IOException
+     */
+
+    protected void parseResponse(String response) {
+        DiscoveryResult result;
+
+        String name = "unknownName";
+        String uuid = "unknownThermostat";
+        String ip = null;
+        String url = null;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.debug("key: {} value: {}.", key, value);
+            switch (key) {
+                case "location":
+                    try {
+                        url = value;
+                        ip = new URL(value).getHost();
```

Comment:
so you only care about the last url and ip that your find and discard the rest?

Reply:
Yes, the discovery message has several lines of output. I only care about the like that has "location" in it. The value will be the thermostat's IP address.